### PR TITLE
ASG interacting implementing a simple method.

### PR DESCRIPTION
**Pull request by ltsampros**
The single method added by asg_instances.py returns true if the current
instance is the first instance of the ASG group when sorted
alphanumerically.

It's a hack accommodating cases where sth needs to run only in one
instance of an ASG group.

### CODE REVIEW

**Comment by sevenmachines** [comment]
@ltsampros 

Think since we're using _grains we should be returning something else rather than true/false? ie, 

   return {"aws": {"asg": {"is_first_in_group": true}}}

or something similar so we get a keyed grain entry

File: _grains/asg_instances.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        conn = boto.ec2.connect_to_region(instance_region)
+        instance_data = conn.get_all_instances(
+            instance_ids=[instance_id])[0].instances[0]
+    except boto.exception.AWSConnectionError as e:
+        log.error("There was a problem collecting instance data, '{}'").format(e.message)
+        return False
+
+    # my autoscaling group
+    asg_group = instance_data.tags['aws:autoscaling:groupName']
+    instance_env = instance_data.tags['Env']
```
Comment (filipposc5) [inline _grains/asg_instances.py]: instance_env is unused .. maybe run flake8 on this to iron out such issues?

Comment (ltsampros) [inline _grains/asg_instances.py]: Agreed.

**Comment by ltsampros** [comment]
@niallcreech Since this is a piece of information that might change, it might make sense to move it to a custom state module. Then we might be able to use the call directly in our state files where a state should be applied only if it is the first node of the group.

**Comment by ltsampros** [comment]
Example usage:

```
# On first node of ASG group
$ python asg.py && echo True
True
# On any other node of ASG group
$ python asg.py && echo True
$
```

One could also summon that from a state but I still don't know how useful or practical it is:

```
{% if salt['asg.is_first_of_asg_group']() == 'True' %}
postgresql-client:
  pkg.installed
{% endif %}
```

**Comment by sevenmachines** [comment]
'if is_first_of_asg_group then enforce this state' is a reasonable workaround for making one of your instances unique in some way. i.e. when you need something enforced on one and only one box. Think its a _util though. Makes little difference but conceptually (according to saltstack) I think thats where it goes.

**Comment by SteveMarshall** [comment]
This is a great feature! Buuuut… `asg_group`? Auto-scaling group group? :disappointed:

**Pull request merged**